integrations/openai-agents-js/src/index.ts

```diff
@@ -0,0 +1,419 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { SpanTypeAttribute } from "@braintrust/core";
+import {
+  Span as BraintrustSpan,
+  startSpan,
+  Experiment,
+  Logger,
+} from "braintrust";
+import type {
+  SpanData,
+  AgentSpanData,
+  FunctionSpanData,
+  GenerationSpanData,
+  ResponseSpanData,
+  HandoffSpanData,
+  CustomSpanData,
+  GuardrailSpanData,
+} from "@openai/agents-core/dist/tracing/spans";
+
+import type { Trace, Span } from "@openai/agents";
+
+// Interface for traces - using the Trace type from @openai/agents
+type AgentsTrace = Trace;
+
+// Interface for spans - using the Span type from @openai/agents
+type AgentsSpan = Span<SpanData>;
+
+// Union types for input/output from different span types
+type SpanInput = string | Array<Record<string, any>> | Record<string, any>[];
+type SpanOutput = string | Array<Record<string, any>> | Record<string, any>;
+
+// Type guard functions
+function isResponseSpanData(spanData: SpanData): spanData is ResponseSpanData {
+  return spanData.type === "response";
+}
+
+function isGenerationSpanData(
+  spanData: SpanData,
+): spanData is GenerationSpanData {
+  return spanData.type === "generation";
+}
+
+function isAgentSpanData(spanData: SpanData): spanData is AgentSpanData {
+  return spanData.type === "agent";
+}
+
+function isFunctionSpanData(spanData: SpanData): spanData is FunctionSpanData {
+  return spanData.type === "function";
+}
+
+function isHandoffSpanData(spanData: SpanData): spanData is HandoffSpanData {
+  return spanData.type === "handoff";
+}
+
+function isGuardrailSpanData(
+  spanData: SpanData,
+): spanData is GuardrailSpanData {
+  return spanData.type === "guardrail";
+}
+
+function isCustomSpanData(spanData: SpanData): spanData is CustomSpanData {
+  return spanData.type === "custom";
+}
+
+function spanTypeFromAgents(span: AgentsSpan): SpanTypeAttribute {
+  const spanType = span.spanData.type;
+
+  if (spanType === "agent" || spanType === "handoff" || spanType === "custom") {
+    return SpanTypeAttribute.TASK;
+  } else if (spanType === "function" || spanType === "guardrail") {
+    return SpanTypeAttribute.TOOL;
+  } else if (spanType === "generation" || spanType === "response") {
+    return SpanTypeAttribute.LLM;
+  } else {
+    return SpanTypeAttribute.TASK;
+  }
+}
+
+function spanNameFromAgents(span: AgentsSpan): string {
+  const spanData = span.spanData;
+
+  if ("name" in spanData && spanData.name) {
+    return spanData.name;
+  }
+
+  switch (spanData.type) {
+    case "generation":
+      return "Generation";
+    case "response":
+      return "Response";
+    case "handoff":
+      return "Handoff";
+    case "agent":
+    case "function":
+    case "guardrail":
+    case "custom":
+      return "name" in spanData && spanData.name ? spanData.name : "Unknown";
+    default:
+      return "Unknown";
+  }
+}
+
+function timestampElapsed(end?: string, start?: string): number | undefined {
```

Comment:
i'm not sure this is a great function name. what about getTimeElapsed? or getInternalMs or something like that